spec/models/customer_spec.rb

```diff
@@ -69,4 +69,45 @@
       expect(c.user).to eq user2
     end
   end
+
+  describe 'scopes' do
+    context 'managed_by' do
+      let!(:admin) { create(:admin_user) }
+      let!(:user) { create(:user) }
+      let!(:enterprise) { create(:enterprise, owner: user) }
+      let!(:customer) { create(:customer, enterprise: enterprise, user: user) }
+      let!(:customer1) { create(:customer, enterprise: enterprise) }
+
+      let!(:user1) { create(:user) }
+      let!(:enterprise1) { create(:enterprise, owner: user1) }
+      let!(:customer2) { create(:customer, enterprise: enterprise1, user: user1) }
+
+      # manager of enterprise1
+      let!(:user2) { create(:user) }
+
+      # user who has edit profile permission on enterprise1
+      let!(:user3) { create(:user) }
+      let!(:enterprise2) { create(:enterprise, owner: user3) }
+
+      before do
+        EnterpriseRole.create!(user: user2, enterprise: enterprise)
+        EnterpriseRelationship.create!(parent: enterprise2, child: enterprise,
+                                       permissions_list: [:edit_profile])
+      end
```

Comment:
I think this is only relevant to the second example below (`user2`), so it  can be moved inside that one, rather than running unnecessarily for each example.

Reply:
`user2` is a manager of `enterprise`.
`user3` has `edit_profile` permission on `enterprise` (I want to make sure that he doesn't have access to `enterprise`'s customers